src/supervisor/metadata-extractor.ts

```diff
@@ -151,6 +154,33 @@ export function isPodAssociatedWithParent(pod: V1Pod): boolean {
     : false;
 }
 
+export function buildNonWorkloadMetadata(
+  kind: string,
+  metadata?: V1ObjectMeta,
+  spec?: V1IngressSpec | V1ServiceSpec,
+): IWorkload {
+  const wl = {
+    type: kind,
+    name: metadata?.name || 'unknown',
+    namespace: metadata?.namespace || 'default',
+    labels: metadata?.labels || {},
```

Comment:
Nit: From L166 - L169, those properties might not require the `|| {}`, as it could have value `undefined`